### ISSUE TITLE
Have code organised in a more 'traditional' project structure

### ISSUE THREAD

**Issue by dvalters** [LSDtopotools/LSDTopoTools2_roadmap#2]
i.e. at the minute the makefiles are burried at the bottom of the directory structure, but we could have something like a typical GNU project like:

```
├── Makefile
├── readme.md
├── requirements.txt
├── bin
├── build
├── docs
├── include
├── src
│   ├── LSDCombobulator.cpp
│   └── LSDFoobarbaz.cpp
└── tests
```
etc

**Comment by simon-m-mudd** [comment]
Yes we absolutely need a structure like this to move all the cpp files into a subdirectory.

**Comment by fclubb** [comment]
Ok just to make sure I understand correctly, we would then have:

- All the executables in the `bin` directory
- All the hpp files in the `include` directory
- All the objects in the `src` directory
- Driver functions: would these go in the `src` directory too?

**Comment by dvalters** [comment]
Yep, I mean it is fairly flexible really - but something that is 'top down' rather than the 'bottom up' layout currently. Sometimes people mix `hpp` and `cpp` in the `src` folder and do away with `include` (maybe simpler). There could be subdirs in the `src` folder as well for drivers (see the CHILD src folder structure for example.)
